upstairs/src/volume.rs

```diff
@@ -1238,14 +1234,130 @@ impl Volume {
                 "Only a single sub_volume is supported"
             )
         }
-        // Sub volumes must all be VolumeConstructionRequest::Region
+
+        // We support replacement for read_only_parent targets with the
+        // same rules as sub_volume targets, however, for any single VCR
+        // update, only one of the sub_volume or read_only_parent targets
+        // can change.
+        let rop_result = {
+            if n_read_only_parent.is_some() {
+                if o_read_only_parent.is_none() {
+                    // We support a VCR where a read_only_parent has gone away,
+                    // but we should never see addition of a read_only_parent
+                    // to a VCR where there was not one originally.
+                    crucible_bail!(
+                        ReplaceRequestInvalid,
+                        "Unexpected read_only_parent on new VCR"
+                    )
+                }
+
+                if o_read_only_parent != n_read_only_parent {
+                    // To get here, both read only parents are populated.
+
+                    let old_rop = o_read_only_parent.unwrap();
+                    let new_rop = n_read_only_parent.unwrap();
+                    // So, the ROPs are different.  Now let's be sure
+                    // they are different in only a target.
+                    Self::compare_vcr_region_for_replacement(&old_rop, &new_rop)
+                } else {
+                    // If the read_only_parents are the same, then this is
+                    // technically an error (from the ROP replacement point of
+                    // view) as it means the targets and generation numbers
+                    // are the same as what we have.
+                    Err(CrucibleError::ReplaceRequestInvalid(
+                        "read_only_parent generation numbers match".to_string(),
+                    ))
+                }
+            } else {
+                // New read_only_parent is none, so the read_only_parents can't
+                // be considered for replacement.
+                Err(CrucibleError::ReplaceRequestInvalid(
+                    "read_only_parent does not exist in replacement"
+                        .to_string(),
+                ))
+            }
+        };
+
+        // Determine the sub_volume difference if there is any.
+        let sub_volume_result = Self::compare_vcr_region_for_replacement(
+            &o_sub_volumes[0],
+            &n_sub_volumes[0],
```

Comment:
This part and probably a whole lot of other areas are going to need revisiting when we move to the future where Nexus is allocating multiple sub-volume entries, but that shouldn't block this PR.

Reply:
Yeah,  I know of a few places as well where we will either explode or report failure if it finds more than one sub-volume.